### ISSUE TITLE
Entry Not Found. Please Check json Response

### ISSUE THREAD

**Issue by nicolas-ceruti** [tussharmate/ios-thermer-custom-schema#4]
I'm using following url to open Thermer App and print: 

thermer://data=%5B%7B%22type%22%3A0%2C%22content%22%3A%22My%20Title%22%2C%22bold%22%3A1%2C%22align%22%3A2%2C%22format%22%3A3%7D%2C%7B%22type%22%3A2%2C%22bold%22%3Anull%2C%22align%22%3A0%2C%22format%22%3Anull%2C%22value%22%3A%221234567890123%22%2C%22height%22%3A50%7D%2C%7B%22type%22%3A3%2C%22bold%22%3Anull%2C%22align%22%3A2%2C%22format%22%3Anull%2C%22value%22%3A%22sample%20qr%20text%22%2C%22height%22%3Anull%2C%22size%22%3A40%7D%2C%7B%22type%22%3A0%2C%22content%22%3A%22%20%22%2C%22bold%22%3A0%2C%22align%22%3A0%7D%2C%7B%22type%22%3A0%2C%22content%22%3A%22This%20text%20has%5Cntwo%20lines%22%2C%22bold%22%3A0%2C%22align%22%3A0%7D%5D

My website can open the app but cant load the data and show the message "Entry Not Found. Please Check json Response"

I tried all the availables options. Someone know why this message keep showing?  I think that this error is showing because the version was updated

**Comment by jbsvector** [comment]
I have the same problem. I have already tested several variations of JSON parameters, but it doesn't work.

An example of a URL used: thermer://?data=%5B%7B%22type%22%3A0%2C%22content%22%3A%22Teste%20de%20Impress%C3%A3o%22%2C%22bold%22%3A1%2C%22align%22%3A1%2C%22format%22%3A2%7D%2C%7B%22type%22%3A0%2C%22content%22%3A%22Ser%C3%A1%20que%20agora%20vai%3F%22%2C%22bold%22%3A0%2C%22align%22%3A0%2C%22format%22%3A1%7D%5D

The App's response is always: "Entry Not Found. Please Check json Response".

The same goes for web printing. (e.g., bprint://https://prod-18.brazilsouth.logic.azure.com/workflows/0bee5f3da606417f90a7b74a0ab53c8d/triggers/manual/paths/invoke?api-version=2016-06-01&sp=%2Ftriggers%2Fmanual%2Frun&sv=1.0&sig=24aju4r36O7Qk6ghItn_C0PvnFJ02bQVJmyDWKtdetM).

**Comment by geekquixotic** [comment]
Okay! I’m happy to say I can now help you both. I spent a **_lot_** of time using the sample code as an experiment and I finally have an answer.

The JSON needs to be a dictionary of dictionaries with each entry being one line. The key for each top level element gets sorted alphabetically by the app for printing.

So the example code gave me something like this:
```
{
  “0” : {
    “type” : 0,
    “content” : “Hello World”,
    “bold” : 1,
    “align” : 1,
    “format” : 2
  },
  “1” : {
    “type”: 0,
    “content” : “This example prints<br>two separate lines.”
  }
}
```

Originally I thought the outer keys had to be sequential numbers - despite the quotes making them strings - and ran into a problem when I had a dozen lines or more and things were out of order. Eventually I figured out the order I was going for and the order I was getting was caused by sorting of strings and  “1,” “10,” and “11” get sorted before “2” alphabetically.

After verifying that I could use any strings I wanted as long as they were in the alphabetical order I wanted, I started using the formatted current time as milliseconds.

And while it doesn’t seem to be an issue in either of your samples, I can say from painful experience that if your text editor uses smart quotes, things break badly.

Good luck!

**Comment by golfeth** [comment]
> I'm using following url to open Thermer App and print:
> 
> thermer://data=%5B%7B%22type%22%3A0%2C%22content%22%3A%22My%20Title%22%2C%22bold%22%3A1%2C%22align%22%3A2%2C%22format%22%3A3%7D%2C%7B%22type%22%3A2%2C%22bold%22%3Anull%2C%22align%22%3A0%2C%22format%22%3Anull%2C%22value%22%3A%221234567890123%22%2C%22height%22%3A50%7D%2C%7B%22type%22%3A3%2C%22bold%22%3Anull%2C%22align%22%3A2%2C%22format%22%3Anull%2C%22value%22%3A%22sample%20qr%20text%22%2C%22height%22%3Anull%2C%22size%22%3A40%7D%2C%7B%22type%22%3A0%2C%22content%22%3A%22%20%22%2C%22bold%22%3A0%2C%22align%22%3A0%7D%2C%7B%22type%22%3A0%2C%22content%22%3A%22This%20text%20has%5Cntwo%20lines%22%2C%22bold%22%3A0%2C%22align%22%3A0%7D%5D
> 
> My website can open the app but cant load the data and show the message "Entry Not Found. Please Check json Response"
> 
> I tried all the availables options. Someone know why this message keep showing? I think that this error is showing because the version was updated

tried everything but can't solved. do you have any idea??

**Comment by johna-freelancer** [comment]
anyone solve this problem?

**Comment by geekquixotic** [comment]
@johna-freelancer 
> anyone solve this problem?

Yes. I posted a solution last September, but I don’t have authority to close the issue. So I’ll take this moment to re-summarize.

The app expects a dictionary of dictionaries rather than a list/array of dictionaries. So the top level’s keys need to be quoted and are processed in alphabetical order. So you don’t want to just use [0,1,2,3,…10,11…] because ‘10’ will get processed before ‘2.’

For my Shortcut I used a timecode for the keys. In this sample below, it’s pulling three reminders and printing them out with a little ASCII frame of sorts around them.

```
{
    "16025670": {
        "type": 0,
        "content": " "
    },
    "16025680": {
        "type": 0,
        "content": "/-<br>|",
        "align": 0,
        "bold": 1
    },
    "16025690": {
        "type": 0,
        "content": "Do the first thing",
        "bold": 1,
        "align": 1,
        "format": 1
    },
    "16025710": {
        "type": 0,
        "content": "03/26",
        "align": 1
    },
    "16025720": {
        "type": 0,
        "content": "|<br>-/",
        "align": 2,
        "bold": 1
    },
    "16025730": {
        "type": 0,
        "content": " "
    },
    "16025750": {
        "type": 0,
        "content": "/-<br>|",
        "align": 0,
        "bold": 1
    },
    "16025760": {
        "type": 0,
        "content": "Also do the second thing",
        "bold": 1,
        "align": 1,
        "format": 1
    },
    "16025770": {
        "type": 0,
        "content": "03/26",
        "align": 1
    },
    "16025790": {
        "type": 0,
        "content": "|<br>-/",
        "align": 2,
        "bold": 1
    },
    "16025800": {
        "type": 0,
        "content": " "
    },
    "16025810": {
        "type": 0,
        "content": "/-<br>|",
        "align": 0,
        "bold": 1
    },
    "16025830": {
        "type": 0,
        "content": "Now do the third thing",
        "bold": 1,
        "align": 1,
        "format": 1
    },
    "16025840": {
        "type": 0,
        "content": "03/26",
        "align": 1
    },
    "16025850": {
        "type": 0,
        "content": "|<br>-/",
        "align": 2,
        "bold": 1
    }
} 
```

Hopefully that answers the question and @tussharmate can close it.

**Comment by johna-freelancer** [comment]
Ohh yeah that works on me! Thanks a lot
